src/main/java/seedu/souschef/logic/parser/contextparser/UniversalParser.java

```diff
@@ -36,23 +38,33 @@
     public Command parseCommand(CommandHistory history, String userInput, Ui ui) throws ParseException {
         final String commandWord = userInput.substring(1);
         switch (commandWord) {
+
+
         case RecipeParser.COMMAND_WORD:
+            EventsCenter.getInstance().post(new BrowserUiChangedEvent("hide"));
```

Comment:
Perform the posting of hide event in the ui.switchToXXXListPanel() method. Same for other context switch.